egs/librispeech/s5/local/run_cleanup_segmentation.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# Copyright 2016  Vimal Manohar
+#           2016  Yiming Wang
+#           2016  Johns Hopkins University (author: Daniel Povey)
+# Apache 2.0
+
+# This script demonstrates how to re-segment training data selecting only the
+# "good" audio that matches the transcripts.
+# The basic idea is to decode with an existing in-domain acoustic model, and a
+# biased language model built from the reference, and then work out the
+# segmentation from a ctm like file.
+
+# For nnet3 and chain results after cleanup, see the scripts in
+# local/nnet3/run_tdnn.sh and local/chain/run_tdnn_6z.sh
+
+# GMM Results for speaker-independent (SI) and speaker adaptive training (SAT) systems on dev and test sets
+# [will add these later].
+
+set -e
+set -o pipefail
+set -u
+
+stage=0
+cleanup_stage=0
+data=data/train_960
+cleanup_affix=cleaned
+srcdir=exp/tri6b
+nj=100
+decode_nj=16
+decode_num_threads=4
+
+. ./path.sh
+. ./cmd.sh
+. utils/parse_options.sh
+
+cleaned_data=${data}_${cleanup_affix}
+
+dir=${srcdir}_${cleanup_affix}_work
+cleaned_dir=${srcdir}_${cleanup_affix}
+
+if [ $stage -le 1 ]; then
+  # This does the actual data cleanup.
+  steps/cleanup/clean_and_segment_data.sh --stage $cleanup_stage --nj $nj --cmd "$train_cmd" \
+    $data data/lang $srcdir $dir $cleaned_data
+fi
+
+if [ $stage -le 2 ]; then
+  steps/align_fmllr.sh --nj $nj --cmd "$train_cmd" \
+    $cleaned_data data/lang $srcdir ${srcdir}_ali_${cleanup_affix}
+fi
+
+if [ $stage -le 3 ]; then
+  steps/train_sat.sh --cmd "$train_cmd" \
+    5000 100000 $cleaned_data data/lang ${srcdir}_ali_${cleanup_affix} ${cleaned_dir}
```

Comment:
Yes, the num-leaves and num-gauss will be the same as in the run.sh


On Mon, Aug 22, 2016 at 4:55 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/librispeech/s5/local/run_cleanup_segmentation.sh
> <https://github.com/kaldi-asr/kaldi/pull/991#discussion_r75756816>:
>
> > +cleaned_dir=${srcdir}_${cleanup_affix}
> > +
> > +if [ $stage -le 1 ]; then
> > +  # This does the actual data cleanup.
> > +  steps/cleanup/clean_and_segment_data.sh --stage $cleanup_stage --nj $nj --cmd "$train_cmd" \
> > +    $data data/lang $srcdir $dir $cleaned_data
> > +fi
> > +
> > +if [ $stage -le 2 ]; then
> > +  steps/align_fmllr.sh --nj $nj --cmd "$train_cmd" \
> > +    $cleaned_data data/lang $srcdir ${srcdir}_ali_${cleanup_affix}
> > +fi
> > +
> > +if [ $stage -le 3 ]; then
> > +  steps/train_sat.sh --cmd "$train_cmd" \
> > +    5000 100000 $cleaned_data data/lang ${srcdir}_ali_${cleanup_affix} ${cleaned_dir}
>
> @danpovey <https://github.com/danpovey> could you let us know how to
> choose the hyper-parameters for the train_sat.sh script. Is it going to be
> the same as the SAT system being built in run.sh ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/991/files/51373685879bf008bd749c8351b4220e24096085#r75756816>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu9Z10HjJhGTyua6x5l6tIGZbTWvbks5qigylgaJpZM4Jpirx>
> .
>